Build warnings when running cmake.

I just did a build of the master branch d1d9839b83800ac04eb4a0bb50d64e2c3efed63c with cmake  following the instructions in the "Compilation" section of README.md, and got the following compiler warnings. (I edited out the cmake progress messages.) This is with gcc 11.2.0 in Linux.

```
/home/warren/repos/git/forks/HiGHS/src/lp_data/Highs.cpp:2842:12: warning: variable 'postsolve_iteration_count' set but not used [-Wunused-but-set-variable]
  HighsInt postsolve_iteration_count = -1;
           ^
1 warning generated.


/home/warren/repos/git/forks/HiGHS/src/mip/HighsDomain.cpp:2403:16: warning: variable 'propnnz' set but not used [-Wunused-but-set-variable]
      HighsInt propnnz = 0;
               ^
/home/warren/repos/git/forks/HiGHS/src/mip/HighsDomain.cpp:2496:18: warning: variable 'propnnz' set but not used [-Wunused-but-set-variable]
        HighsInt propnnz = 0;
                 ^
2 warnings generated.


/home/warren/repos/git/forks/HiGHS/src/mip/HighsCliqueTable.cpp:1667:18: warning: variable 'numDel' set but not used [-Wunused-but-set-variable]
        HighsInt numDel = 0;
                 ^
1 warning generated.


/home/warren/repos/git/forks/HiGHS/src/presolve/HighsLpPropagator.cpp:485:14: warning: variable 'propnnz' set but not used [-Wunused-but-set-variable]
    HighsInt propnnz = 0;
             ^
1 warning generated.


/home/warren/repos/git/forks/HiGHS/src/presolve/Presolve.cpp:188:12: warning: variable 'total_rows' set but not used [-Wunused-but-set-variable]
  HighsInt total_rows = 0;
           ^
/home/warren/repos/git/forks/HiGHS/src/presolve/Presolve.cpp:1027:12: warning: variable 'cntEl' set but not used [-Wunused-but-set-variable]
  HighsInt cntEl = 0;
           ^
2 warnings generated.


In file included from /home/warren/repos/git/forks/HiGHS/src/qpsolver/quass.cpp:13:
In file included from /home/warren/repos/git/forks/HiGHS/src/qpsolver/feasibility.hpp:7:
/home/warren/repos/git/forks/HiGHS/src/qpsolver/feasibility_quass.hpp:31:13: warning: empty parentheses interpreted as a function declaration [-Wvexing-parse]
 Basis basis();
            ^~
/home/warren/repos/git/forks/HiGHS/src/qpsolver/feasibility_quass.hpp:31:13: note: replace parentheses with an initializer to declare a variable
 Basis basis();
            ^~
            {}
1 warning generated.



/home/warren/repos/git/forks/HiGHS/src/util/HighsUtils.cpp:538:12: warning: variable 'sumK' set but not used [-Wunused-but-set-variable]
  HighsInt sumK = 0;
           ^
/home/warren/repos/git/forks/HiGHS/src/util/HighsUtils.cpp:541:12: warning: variable 'sumPct' set but not used [-Wunused-but-set-variable]
  HighsInt sumPct = 0;
           ^
/home/warren/repos/git/forks/HiGHS/src/util/HighsUtils.cpp:722:10: warning: variable 'sum_percentage' set but not used [-Wunused-but-set-variable]
  double sum_percentage = 0;
         ^
3 warnings generated.
```

@feldmeier : Branch fix-836 deals with trivial cases (other than those in the MIP solver), but what about 

> In file included from /home/warren/repos/git/forks/HiGHS/src/qpsolver/feasibility.hpp:7:
> I/home/warren/repos/git/forks/HiGHS/src/qpsolver/feasibility_quass.hpp:31:13: warning: empty parentheses > > interpreted as a function declaration [-Wvexing-parse]
>  Basis basis();
>              ^~
>  /home/warren/repos/git/forks/HiGHS/src/qpsolver/feasibility_quass.hpp:31:13: note: replace parentheses with an initializer to declare a variable
>   Basis basis();
>              ^~
>              {}
>  1 warning generated.

Could this ever be a problem?

--- a/src/lp_data/Highs.cpp
+++ b/src/lp_data/Highs.cpp
@@ -2839,7 +2839,6 @@ HighsStatus Highs::callRunPostsolve(const HighsSolution& solution,
   presolve_.data_.recovered_solution_ = solution;
   presolve_.data_.recovered_basis_ = basis;
 
-  HighsInt postsolve_iteration_count = -1;
   HighsPostsolveStatus postsolve_status = runPostsolve();
   if (postsolve_status == HighsPostsolveStatus::kSolutionRecovered) {
     highsLogDev(options_.log_options, HighsLogType::kVerbose,
@@ -2874,14 +2873,12 @@ HighsStatus Highs::callRunPostsolve(const HighsSolution& solution,
     // Set up the iteration count and timing records so that
     // adding the corresponding values after callSolveLp gives
     // difference
-    postsolve_iteration_count = -info_.simplex_iteration_count;
     timer_.start(timer_.solve_clock);
     call_status = callSolveLp(
         incumbent_lp,
         "Solving the original LP from the solution after postsolve");
     timer_.stop(timer_.solve_clock);
     // Determine the iteration count and timing records
-    postsolve_iteration_count += info_.simplex_iteration_count;
     return_status = interpretCallStatus(options_.log_options, call_status,
                                         return_status, "callSolveLp");
     // Recover the options

--- a/src/lp_data/Highs.cpp
+++ b/src/lp_data/Highs.cpp
@@ -2870,15 +2870,14 @@ HighsStatus Highs::callRunPostsolve(const HighsSolution& solution,
     // Scrap the EKK data from solving the presolved LP
     ekk_instance_.invalidate();
     ekk_instance_.lp_name_ = "Postsolve LP";
-    // Set up the iteration count and timing records so that
-    // adding the corresponding values after callSolveLp gives
-    // difference
+    // Set up the timing record so that adding the corresponding
+    // values after callSolveLp gives difference
     timer_.start(timer_.solve_clock);
     call_status = callSolveLp(
         incumbent_lp,
         "Solving the original LP from the solution after postsolve");
+    // Determine the timing record
     timer_.stop(timer_.solve_clock);
-    // Determine the iteration count and timing records
     return_status = interpretCallStatus(options_.log_options, call_status,
                                         return_status, "callSolveLp");
     // Recover the options

--- a/src/presolve/Presolve.cpp
+++ b/src/presolve/Presolve.cpp
@@ -1024,9 +1024,9 @@ void Presolve::UpdateMatrixCoeffDoubletonEquationXnonZero(
 }
 
 void Presolve::trimA() {
-  HighsInt cntEl = 0;
-  for (HighsInt j = 0; j < numCol; ++j)
-    if (flagCol.at(j)) cntEl += nzCol.at(j);
+  //  HighsInt cntEl = 0;
+  //  for (HighsInt j = 0; j < numCol; ++j)
+  //    if (flagCol.at(j)) cntEl += nzCol.at(j);
 
   vector<pair<HighsInt, size_t>> vp;
   vp.reserve(numCol);

--- a/src/util/HighsUtils.cpp
+++ b/src/util/HighsUtils.cpp
@@ -535,16 +535,12 @@ void analyseMatrixSparsity(const HighsLogOptions& log_options,
   }
   HighsInt cat = maxCat;
   if (colCatK[cat]) lastRpCat = cat;
-  HighsInt sumK = 0;
   HighsInt pct;
   double v;
-  HighsInt sumPct = 0;
   for (HighsInt cat = 0; cat < lastRpCat; cat++) {
-    sumK += colCatK[cat];
     v = 100 * colCatK[cat];
     v = v / numCol + 0.5;
     pct = v;
-    sumPct += pct;
     highsLogDev(log_options, HighsLogType::kInfo,
                 "%12" HIGHSINT_FORMAT " (%3" HIGHSINT_FORMAT
                 "%%) columns of count in [%3" HIGHSINT_FORMAT
@@ -553,11 +549,9 @@ void analyseMatrixSparsity(const HighsLogOptions& log_options,
   }
 
   cat = lastRpCat;
-  sumK += colCatK[cat];
   v = 100 * colCatK[cat];
   v = v / numCol + 0.5;
   pct = v;
-  sumPct += pct;
   if (cat == maxCat) {
     highsLogDev(log_options, HighsLogType::kInfo,
                 "%12" HIGHSINT_FORMAT " (%3" HIGHSINT_FORMAT
@@ -580,16 +574,12 @@ void analyseMatrixSparsity(const HighsLogOptions& log_options,
   }
   cat = maxCat;
   if (rowCatK[cat]) lastRpCat = cat;
-  sumK = 0;
   pct = 0;
   v = 0;
-  sumPct = 0;
   for (HighsInt cat = 0; cat < lastRpCat; cat++) {
-    sumK += rowCatK[cat];
     v = 100 * rowCatK[cat];
     v = v / numRow + 0.5;
     pct = v;
-    sumPct += pct;
     highsLogDev(log_options, HighsLogType::kInfo,
                 "%12" HIGHSINT_FORMAT " (%3" HIGHSINT_FORMAT
                 "%%)    rows of count in [%3" HIGHSINT_FORMAT
@@ -598,11 +588,9 @@ void analyseMatrixSparsity(const HighsLogOptions& log_options,
   }
 
   cat = lastRpCat;
-  sumK += rowCatK[cat];
   v = 100 * rowCatK[cat];
   v = v / numRow + 0.5;
   pct = v;
-  sumPct += pct;
   if (cat == maxCat) {
     highsLogDev(log_options, HighsLogType::kInfo,
                 "%12" HIGHSINT_FORMAT " (%3" HIGHSINT_FORMAT
@@ -719,7 +707,6 @@ bool logValueDistribution(const HighsLogOptions& log_options,
   bool not_reported_ones = true;
   HighsInt sum_count =
       value_distribution.num_zero_ + value_distribution.num_one_;
-  double sum_percentage = 0;
   const double min_value = value_distribution.min_value_;
   for (HighsInt i = 0; i < num_count + 1; i++)
     sum_count += value_distribution.count_[i];
@@ -752,7 +739,6 @@ bool logValueDistribution(const HighsLogOptions& log_options,
   HighsInt count = value_distribution.num_zero_;
   if (count) {
     percentage = doublePercentage(count, sum_count);
-    sum_percentage += percentage;
     int_percentage = percentage;
     highsLogDev(log_options, HighsLogType::kInfo,
                 "%12" HIGHSINT_FORMAT " %svalues (%3" HIGHSINT_FORMAT
@@ -763,7 +749,6 @@ bool logValueDistribution(const HighsLogOptions& log_options,
   count = value_distribution.count_[0];
   if (count) {
     percentage = doublePercentage(count, sum_count);
-    sum_percentage += percentage;
     int_percentage = percentage;
     highsLogDev(log_options, HighsLogType::kInfo,
                 "%12" HIGHSINT_FORMAT " %svalues (%3" HIGHSINT_FORMAT
@@ -785,7 +770,6 @@ bool logValueDistribution(const HighsLogOptions& log_options,
       count = value_distribution.num_one_;
       if (count) {
         percentage = doublePercentage(count, sum_count);
-        sum_percentage += percentage;
         int_percentage = percentage;
         highsLogDev(log_options, HighsLogType::kInfo,
                     "%12" HIGHSINT_FORMAT " %svalues (%3" HIGHSINT_FORMAT
@@ -804,7 +788,6 @@ bool logValueDistribution(const HighsLogOptions& log_options,
     count = value_distribution.count_[i];
     if (count) {
       percentage = doublePercentage(count, sum_count);
-      sum_percentage += percentage;
       int_percentage = percentage;
       highsLogDev(log_options, HighsLogType::kInfo,
                   "%12" HIGHSINT_FORMAT " %svalues (%3" HIGHSINT_FORMAT
@@ -828,7 +811,6 @@ bool logValueDistribution(const HighsLogOptions& log_options,
     count = value_distribution.num_one_;
     if (count) {
       percentage = doublePercentage(count, sum_count);
-      sum_percentage += percentage;
       int_percentage = percentage;
       highsLogDev(log_options, HighsLogType::kInfo,
                   "%12" HIGHSINT_FORMAT " %svalues (%3" HIGHSINT_FORMAT
@@ -847,7 +829,6 @@ bool logValueDistribution(const HighsLogOptions& log_options,
   count = value_distribution.count_[num_count];
   if (count) {
     percentage = doublePercentage(count, sum_count);
-    sum_percentage += percentage;
     int_percentage = percentage;
     highsLogDev(log_options, HighsLogType::kInfo,
                 "%12" HIGHSINT_FORMAT " %svalues (%3" HIGHSINT_FORMAT
@@ -867,7 +848,6 @@ bool logValueDistribution(const HighsLogOptions& log_options,
     count = value_distribution.num_one_;
     if (count) {
       percentage = doublePercentage(count, sum_count);
-      sum_percentage += percentage;
       int_percentage = percentage;
       highsLogDev(log_options, HighsLogType::kInfo,
                   "%12" HIGHSINT_FORMAT " %svalues (%3" HIGHSINT_FORMAT

